bugbug/feature_cleanup.py

```diff
@@ -174,18 +174,22 @@ def __init__(self):
                 ],
             ),
             ("spec", ["spec", "specification"]),
-        ]
-        self.pattern = {}
-        for synonym_group, synonym_list in synonyms:
-            self.pattern[synonym_group] = re.compile(
-                "|".join(rf"\b{synonym}\b" for synonym in synonym_list),
-                flags=re.IGNORECASE,
-            )
+        )
+        self.synonyms_dict = {
+            synonym: synonym_group
+            for synonym_group, synonym_list in synonyms
+            for synonym in synonym_list
+        }
+        self.pattern = re.compile(
+            r"|".join(rf"\b{synonym}\b" for synonym in self.synonyms_dict.keys()),
+            flags=re.IGNORECASE,
+        )
 
     def __call__(self, text):
-        for synonym_group in self.pattern:
-            text = self.pattern[synonym_group].sub(synonym_group, text)
-        return text
+        def replacement(match):
+            return self.synonyms_dict[match.group(0).lower()]
```

Comment:
nit: Moving `replacement()` to be a private method could have a slight performance improvement.

Reply:
It did get to 4.2 seconds. I was expecting it to be slightly better but thought it would have been almost not visible!